src/tribler/gui/start_gui.py

```diff
@@ -41,14 +42,18 @@ def run_gui(api_port, api_key, root_state_dir, parsed_args):
 
     # Enable tracer using commandline args: --trace-debug or --trace-exceptions
     trace_logger = check_and_enable_code_tracing('gui', root_state_dir)
-    try:
-        enable_fault_handler(root_state_dir)
-        # Exit if we cant read/write files, etc.
-        check_environment()
-        check_free_space()
 
+    enable_fault_handler(root_state_dir)
+    # Exit if we cant read/write files, etc.
+    check_environment()
+    check_free_space()
+
+    process_locker = ProcessLocker(root_state_dir, ProcessKind.GUI)
+    another_process_is_active = not process_locker.current_process.active
+    set_global_process_locker(process_locker)   # to be able to add information about exception to the process info
+    try:
         app_name = os.environ.get('TRIBLER_APP_NAME', 'triblerapp')
-        app = TriblerApplication(app_name, sys.argv)
+        app = TriblerApplication(app_name, sys.argv, another_process_is_active)
```

Comment:
I guess I understand your intention to pass the argument `another_process_is_active` to `TriblerApplication` but maybe it is not worth it in the end.

It is used only as:

```python
  ...
  if app.is_running():
            # if an application is already running, then send the command line
            # argument to it and close the current instance
            ...
```

That is why I would suggest removing the passing `another_process_is_active` logic completely and using `not process_locker.current_process.active` instead:

```python
  ...
  if another_process_is_active
            # if an application is already running, then send the command line
            # argument to it and close the current instance
            ...
```

Reply:
I changed the logic to make it more straightforward:
```python
current_process_is_primary = process_manager.current_process.become_primary()
...
app = TriblerApplication(app_name, sys.argv, start_local_server=current_process_is_primary)
...
if not current_process_is_primary and app.connected_to_previous_instance:
    logger.info('GUI Application is already running. Passing a torrent file path to it.')
    ...
```